src/rest.rs

```diff
@@ -1,22 +1,20 @@
 use crate::error::BrigadeError;
-use hyper::Method;
-use reqwest::Response;
-use serde::{Deserialize, Serialize};
-use serde_with::*;
+use reqwest::{IntoUrl, Method, RequestBuilder};
 
 #[derive(Debug, Clone)]
 pub struct ClientConfig {
     pub allow_insecure_connections: bool,
 }
 
 impl ClientConfig {
-    pub fn default() -> Self {
```

Comment:
Nit: so in this case, default is fine as you are passing no parameters (clippy might even tell you this if you run it, but I can't remember). If you are planning on allowing people to pass params to this in the future, `new` would be correct. However, I would recommend rolling up this and the params in the `Client::new` function to be a builder. So if you are specifying all the options:

```rust
let client = ClientBuilder::new("http://foo.bar").allow_insecure_connections().token("deadbeef").build();
```

Basic auth would look like this (and could do the auth flow for you):

```rust
let client = ClientBuilder::new("https://foo.bar").auth(user, pass).build();
```

And root auth would look like this:

```rust
let client = ClientBuilder::new("https://foo.bar").root_auth(pass).build();
```

Those last auth options would move the auth stuff into the top level client where it can be reused for all clients. However, all of this is probably best done in a follow up